### PR TITLE
New function `ValueFrom` that takes a Go value and populates a compatible `attr.Value`, given a descriptive `attr.Type`

### PR DESCRIPTION
**Pull request by megan07**
fixes https://github.com/hashicorp/terraform-plugin-framework/issues/323

### CODE REVIEW

**Comment by detro** [comment]
Testing this locally now

**Review by detro — APPROVED** [review]

File: tfsdk/value_from_test.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+type person struct {
+	Name    types.String `tfsdk:"name"`
+	Age     types.Int64  `tfsdk:"age"`
+	OptedIn types.Bool   `tfsdk:"opted_in"`
+	Address types.List   `tfsdk:"address"`
```
Comment (bendbennett) [inline tfsdk/value_from_test.go:18]: Is it worth testing other types (e.g., `types.Map`), along the lines of
```go
	Address types.Map    `tfsdk:"address"`
```

Then using the following in `personAttrTypes`:
```go
		"address": types.MapType{
			ElemType: types.StringType,
		},
```

Populating `mrX` with:
```go
		Address: types.Map{
			ElemType: types.StringType,
			Elems: map[string]attr.Value{
				"number": types.String{Value: "1"},
				"street": types.String{Value: "Beckford Close"},
				"town":   types.String{Value: "Gotham"},
			},
		},
```

And finally testing `expectedMrXObj`:
```go
			"address": types.Map{
				ElemType: types.StringType,
				Elems: map[string]attr.Value{
					"number": types.String{Value: "1"},
					"street": types.String{Value: "Beckford Close"},
					"town":   types.String{Value: "Gotham"},
				},
			},
```

**Review by bendbennett — APPROVED** [review]
Nice. Just a minor comment.

File: tfsdk/value_from_test.go (hunk 1 above)
Comment (detro) [reply tfsdk/value_from_test.go]: Just added it ;)

**Pull request merged**